### PR TITLE
Autoyast should boot from local disk on migration test for aarch64

### PR DESCRIPTION
**Pull request by lemon-suse**
We should consider the different boot locations for aarch64. On aarch64, it will firstly on the 'inst-bootmenu', then up to 'inst-bootmenu-boot-harddisk' to boot from hard disk.

- Related ticket: https://progress.opensuse.org/issues/46124
- Needles: N/A
- Verification run: http://openqa-apac1.suse.de/tests/3340#

### CODE REVIEW

**Comment by lemon-suse** [comment]
Hi Oliver, I just checked the migration flow on aarch64, it will reboot after the installation for migration and it will default on 'installation' for bootloader, we must boot on local disk so I added these steps copy from function wait_boot. In fact, we haven't added migration autoyast testcases on aarch64 before, so I do think here we need to do this modification. I think there is difference between the process of migration and installation.  
Please refer to https://openqa.suse.de/tests/2399270#step/grub_test/1 , this is also reboot after the installation of migration and the bootloader will default on 'installation', so here we need the same action to boot from hard disk. WDYT?

**Comment by okurz** [comment]
Yes I think you are right that the preselected entry could be different for the case of installation or migration but why is the subject line stating "Boot from local disk on **installation** on aarch64 by autoyast". I am confused now. Is this about *installation* or *migration*? I assume migration using autoyast.

What do you think about using the method `wait_boot` instead of copying it into the autoyast installation file? This should make long-term code maintenance easier. You could also introduce an option to wait_boot when you want to stop in the grub menu if that helps?

**Comment by okurz** [comment]
@rwx788 as one of the main contributors to the test modules "autoyast/installation" I am inviting you for review as well.

**Comment by lemon-suse** [comment]
> Yes I think you are right that the preselected entry could be different for the case of installation or migration but why is the subject line stating "Boot from local disk on **installation** on aarch64 by autoyast". I am confused now. Is this about _installation_ or _migration_? I assume migration using autoyast.
Sorry for the confusing title, the issue is after migration process and reboot, the preselected entry on aarch64 is different from other arches, so we need to modify code to suit it. I'll thinking a more suitable title for it.
> 
> What do you think about using the method `wait_boot` instead of copying it into the autoyast installation file? This should make long-term code maintenance easier. You could also introduce an option to wait_boot when you want to stop in the grub menu if that helps?
Yes, I will think a good method to reuse the method 'wait_boot' in autoyast code.

**Comment by okurz** [comment]
your quoting seems to be a bit off but I think I got the message :)

**Comment by rwx788** [comment]
Hey! @lemon-suse Am I right that you haven't yet introduced changes we've agreed on when you were visiting us? To extract that method and reuse single implementation. #6747 reminded me about it as it's same as what you've added here, just doesn't include more complex cases for decrypting partitions.

File: tests/autoyast/installation.pm [hunk 1]
```diff
@@ -102,6 +104,23 @@ sub verify_timeout_and_check_screen {
     return check_screen $needles, $check_time;
 }
 
+sub wait_boot_on_local_disk {
+   send_key_until_needlematch 'inst-bootmenu-boot-harddisk', 'up';
```
Comment (rwx788) [inline tests/autoyast/installation.pm]: Once again, we have 100% same code in https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/lib/opensusebasetest.pm#L370 
Could you please put this method in opensusebasetest and call it in L#374 and in this module.

Comment (lemon-suse) [reply tests/autoyast/installation.pm]: Updated code following your comments, please have a look.

Comment (rwx788) [reply tests/autoyast/installation.pm]: LGTM, please, resolve conflicts and we can merge.

File: lib/opensusebasetest.pm [hunk 2]
```diff
@@ -583,6 +566,26 @@ sub wait_boot {
...
+sub wait_boot_on_local_disk {
+   # assuming the cursor is on 'installation' by default and 'boot from
+   # harddisk' is above
+   send_key_until_needlematch 'inst-bootmenu-boot-harddisk', 'up';
+   boot_local_disk;
+   my @tags = qw(grub2 tianocore-mainmenu);
+   push @tags, 'encrypted-disk-password-prompt' if (get_var('ENCRYPT'));
+
+   check_screen(\@tags, 15)
+              || die 'neither grub2 nor tianocore-mainmenu needles found';
```
Comment (okurz) [inline lib/opensusebasetest.pm]: Please replace that by just `assert_screen`

File: lib/opensusebasetest.pm [hunk 3]
```diff
@@ -583,6 +566,26 @@ sub wait_boot {
     $self->{in_wait_boot} = 0;
 }
 
+sub wait_boot_on_local_disk {
```
Comment (okurz) [inline lib/opensusebasetest.pm]: I am not sure about the name. We do not actually wait for the boot to complete as we are waiting for the grub-menu in the end, right? I think we need to find a better name for the method. Can you please also add a doc-string same as for other methods, e.g. as for `wait_boot`

File: tests/autoyast/installation.pm [hunk 4]
```diff
@@ -22,6 +22,7 @@ use testapi;
 use utils;
 use power_action_utils 'prepare_system_shutdown';
 use version_utils qw(is_sle is_caasp is_released);
+use opensusebasetest;
```
Comment (rwx788) [inline tests/autoyast/installation.pm]: You can actually skip that import and use `$self` as all test modules are inherited from opensusebasetest.

File: lib/opensusebasetest.pm (hunk 2 above)
Comment (lemon-suse) [reply lib/opensusebasetest.pm]: Done.

File: lib/opensusebasetest.pm (hunk 3 above)
Comment (lemon-suse) [reply lib/opensusebasetest.pm]: I change the name to wait_grub_to_boot_on_local_disk. WDYT?

File: tests/autoyast/installation.pm (hunk 4 above)
Comment (lemon-suse) [reply tests/autoyast/installation.pm]: Updated code, please have a look.

File: lib/opensusebasetest.pm [hunk 5]
```diff
@@ -398,6 +381,26 @@ sub wait_grub {
     mutex_wait 'support_server_ready' if get_var('USE_SUPPORT_SERVER');
 }
 
+=head2 wait_grub_to_boot_on_local_disk
```
Comment (rwx788) [inline lib/opensusebasetest.pm]: Seems that you've missed putting the description and closing section.
It should be like:
```perl
=head2 wait_grub_to_boot_on_local_disk

    wait_grub_to_boot_on_local_disk

<Description goes here>

=cut
```

Comment (lemon-suse) [reply lib/opensusebasetest.pm]: Updated, please have a look.

**Pull request merged**